### PR TITLE
Inventory equip

### PR DESCRIPTION
**Pull request by LethalGhost**
Update `equip` part of inventory code. Don't confuse with `hands` part (put_in_hands, put_in_acvive_hand, drop_l_hand).
Basic procs:
- `equip_in_slot` - UNSAFE make all "equip works"
- `equip_in_slot_if_possible` - basic proc if you wanna propertly equip item
- `equip_in_slot_or_del` - equip_in_slot_if_possible + del if can't be equipped
- `equip_to_appropriate_slot` - equip_in_slot_if_possible + autoselect slot

BTW, touch hands, helpers parts.
All common changes and ideas described in [`code/modules/mob/inventory/_docs.dm`](https://github.com/discordia-space/CEV-Eris/blob/35a7ba3f6741812cf00044042b2879adb1141a5c/code/modules/mob/inventory/_docs.dm)
Check it (and it's changes if you wanna know what changes better).

Important change:
Remove `/obj/item/proc/mob_can_equip(Mob...)`
It code was splited into `mob/can_equip(Item, slot...)`, `Item/can_be_equipped(Mob, slot...)` and slot datums `/can_equip(Mob, Item...)`

Side changes:
move equip sound from `equip_to_slot_if_possible` to `pre_equip`
remove `can_pickup` proc, all functionality kind of included in `equip_to_slot_if_possible`

### CODE REVIEW

**Comment by LethalGhost** [comment]
There will be tiny update for `/mob/living/carbon/human/equip_to_slot(...)` in `code/modules/mob/living/carbon/human/inventory.dm`

**Comment by seekin4u** [comment]
Мой звуко-ПР интегрирован, отлично

File: code/__defines/items_clothing.dm [hunk 1]
```diff
@@ -78,6 +78,7 @@
 #define FLEXIBLEMATERIAL           0x20 // At the moment, masks with this flag will not prevent eating even if they are covering your face.
 #define COVER_PREVENT_MANIPULATION 0x40 // Only clothing with this flag will prevent manipulation under it. Its for space suits and such, unlike from usual Bay12 rules of clothing manipulation.
 #define DRAG_AND_DROP_UNEQUIP      0x80 // Allow you put intems in hands with drag and drop
+#define LOUDLY_EQUIPEMENT          0x100// Play sound when equipped/unequipped
```
Comment (NanakoAC) [inline code/__defines/items_clothing.dm]: This needs to be renamed something better
try EQUIP_SOUNDS

File: code/game/objects/items/_inventory.dm [hunk 2]
```diff
@@ -44,115 +40,9 @@
...
-				if(!disable_warning)
-					H << SPAN_WARNING("You need a jumpsuit before you can attach this [name].")
-				return 0
-			var/obj/item/clothing/under/uniform = H.w_uniform
-			if(uniform.accessories.len && !uniform.can_attach_accessory(src))
-				if (!disable_warning)
-					H << SPAN_WARNING("You already have an accessory of this type attached to your [uniform].")
-				return 0
-	return 1
+/obj/item/proc/can_be_equipped(mob/Mob, slot, disable_warning = FALSE)
```
Comment (NanakoAC) [inline code/game/objects/items/_inventory.dm]: I think the title of this was better before, mob_can_equip is more descriptive

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 3]
```diff
@@ -207,130 +169,97 @@ This saves us from having to call add_fingerprint() any time something is put in
...
 			src.head = W
 			if(head.flags_inv & (BLOCKHAIR|BLOCKHEADHAIR|HIDEMASK))
 				update_hair(redraw_mob)	//rebuild hair
 				update_inv_ears(0)
 				update_inv_wear_mask(0)
-			W.equipped(src, slot)
-			update_inv_head(redraw_mob)
 		if(slot_shoes)
 			src.shoes = W
-			W.equipped(src, slot)
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/inventory.dm]: You've removed a lot of calls to equipped proc here. Have they been moved elsewhere?

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 4]
```diff
@@ -207,130 +169,97 @@ This saves us from having to call add_fingerprint() any time something is put in
...
 		if(slot_wear_suit)
 			src.wear_suit = W
 			if(wear_suit.flags_inv & HIDESHOES)
 				update_inv_shoes(0)
-			W.equipped(src, slot)
-			update_inv_wear_suit(redraw_mob)
 		if(slot_w_uniform)
 			src.w_uniform = W
-			W.equipped(src, slot)
-			update_inv_w_uniform(redraw_mob)
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/inventory.dm]: why remove all these update_inv procs? how is the updating handled now?

**Comment by LethalGhost** [comment]
@NanakoAC, `equipped(...)` and `update_SLOT__icon` now handled with slot datums in `equip_to_slot`.

**Comment by NanakoAC** [comment]
High risk, unadddressed questions. May be best to delay this one for now

**Comment by NanakoAC** [comment]
our timing was off, lemme see again

**Comment by LethalGhost** [comment]
@NanakoAC [here](https://github.com/discordia-space/CEV-Eris/pull/2176/files#diff-bd8addc57db9e395a3207b90acedec2dR172)

**Pull request merged**